phoenix/src/main/java/org/apache/zeppelin/phoenix/PhoenixInterpreter.java

```diff
@@ -0,0 +1,234 @@
+        .add(PHOENIX_JDBC_USER, DEFAULT_JDBC_USER, "The Phoenix user")
+        .add(PHOENIX_JDBC_PASSWORD, DEFAULT_JDBC_PASSWORD, "The password for the Phoenix user")
+        .add(PHOENIX_MAX_RESULT, DEFAULT_MAX_RESULT, "Max number of SQL results to display.")
+        .add(PHOENIX_JDBC_DRIVER_NAME, DEFAULT_JDBC_DRIVER_NAME, "Phoenix Driver classname.")
+        .build()
+    );
+  }
+
+  public PhoenixInterpreter(Properties property) {
+    super(property);
+  }
+
+  @Override
+  public void open() {
+    logger.info("Jdbc open connection called!");
+    close();
+
+    try {
+      Class.forName(getProperty(PHOENIX_JDBC_DRIVER_NAME));
+
+      maxResult = Integer.valueOf(getProperty(PHOENIX_MAX_RESULT));
+      jdbcConnection = DriverManager.getConnection(
+        getProperty(PHOENIX_JDBC_URL),
+        getProperty(PHOENIX_JDBC_USER),
+        getProperty(PHOENIX_JDBC_PASSWORD)
+      );
+      exceptionOnConnect = null;
+      logger.info("Successfully created Jdbc connection");
+    }
+    catch (ClassNotFoundException | SQLException e) {
+      logger.error("Cannot open connection", e);
+      exceptionOnConnect = e;
+    }
+  }
+
+  @Override
+  public void close() {
+    logger.info("Jdbc close connection called!");
+
+    try {
+      if (getJdbcConnection() != null) {
+        getJdbcConnection().close();
+      }
+    } catch (SQLException e) {
+      logger.error("Cannot close connection", e);
+    }
+    finally {
+      exceptionOnConnect = null;
+    }
+  }
+
+  private String clean(boolean isExplain, String str){
+    return (isExplain || str == null) ? str : str.replace(TAB, WS).replace(NEWLINE, WS);  
+  }
+
+  private InterpreterResult executeSql(String sql) {
+    try {
+      if (exceptionOnConnect != null) {
+        return new InterpreterResult(Code.ERROR, exceptionOnConnect.getMessage());
+      }
+
+      currentStatement = getJdbcConnection().createStatement();
+
+      boolean isExplain = StringUtils.containsIgnoreCase(sql, EXPLAIN_PREDICATE);
+      StringBuilder msg = (isExplain) ? new StringBuilder() : new StringBuilder(TABLE_MAGIC_TAG);
+
+      try {
+        if (currentStatement.execute(sql)){ //If query had results
+          ResultSet res = currentStatement.getResultSet();
+
+          //Append column names
+          ResultSetMetaData md = res.getMetaData();
+          String row = clean(isExplain, md.getColumnName(1));
+          for (int i = 2; i < md.getColumnCount() + 1; i++)
+            row += TAB + clean(isExplain, md.getColumnName(i));
+          msg.append(row + NEWLINE);
+
+          //Append rows
+          int rowCount = 0;
+          while (res.next() && rowCount < getMaxResult()) {
+            row = clean(isExplain, res.getString(1));
+            for (int i = 2; i < md.getColumnCount() + 1; i++)
+              row += TAB + clean(isExplain, res.getString(i));
+            msg.append(row + NEWLINE);
+            rowCount++;
+          }
+          res.close();
```

Comment:
It is safer to close the ResultSet in the `final` block. Right now it seems it is not being closed in the case of false execute result.